tests/testhelpers.py

```diff
@@ -1,257 +1,209 @@
-
-#         self.realm.exchange.step()
-
-#         for ent_id, ent in self.dead.items():
-#             #if ent.agent.scripted:
-#             #    continue
-#             rewards[ent.ent_id], infos[ent.ent_id] = self.reward(ent)
-
-#             dones[ent.ent_id] = False #TODO: Is this correct behavior?
-
-#             #obs[ent.ent_id]     = self.dummy_ob
-
-#         #Pettingzoo API
-#         self.agents = list(self.realm.players.keys())
-
-#         self.obs = obs
-#         return obs, rewards, dones, infos
-
-
-# class TestConfig(nmmo.config.Small, nmmo.config.AllGameSystems):
-
-#     __test__ = False
-
-#     RENDER = False
-#     SPECIALIZE = True
-#     PLAYERS = [
-#             baselines.Fisher, baselines.Herbalist, baselines.Prospector, baselines.Carver, baselines.Alchemist,
-#             baselines.Melee, baselines.Range, baselines.Mage]
+import numpy as np
+from copy import deepcopy
+
+import nmmo
+
+from scripted import baselines
+
+
+# this function can be replaced by assertDictEqual
+# but might be still useful for debugging
+def are_actions_equal(source_atn, target_atn, debug=True):
+
+  # compare the numbers and player ids
+  player_src = list(source_atn.keys())
+  player_tgt = list(target_atn.keys())
+  if player_src != player_tgt:
+    if debug:
+      print("players don't match")
+    return False
+
+  # for each player, compare the actions
+  for ent_id in player_src:
+    atn1 = source_atn[ent_id]
+    atn2 = target_atn[ent_id]
+
+    if list(atn1.keys()) != list(atn2.keys()):
+      if debug:
+        print("action keys don't match. player:", ent_id)
+      return False
+
+    for atn, args in atn1.items():
+      if atn2[atn] != args:
+        if debug:
+          print("action args don't match. player:", ent_id, ", action:", atn)
+        return False
+
+  return True
+
+
+# this function CANNOT be replaced by assertDictEqual
+def are_observations_equal(source_obs, target_obs, debug=True):
+
+  keys_src = list(source_obs.keys())
+  keys_obs = list(target_obs.keys())
+  if keys_src != keys_obs:
+    if debug:
+      print("entities don't match")
+    return False
+
+  for k in keys_src:
+    ent_src = source_obs[k]
+    ent_tgt = target_obs[k]
+    if list(ent_src.keys()) != list(ent_tgt.keys()):
+      if debug:
+        print("entries don't match. key:", k)
+      return False
+
+    obj = ent_src.keys()
+    for o in obj:
+      obj_src = ent_src[o]
+      obj_tgt = ent_tgt[o]
+      if np.sum(obj_src != obj_tgt) > 0:
+        if debug:
+          print("objects don't match. key:", k, ', obj:', o)
+        return False
+
+  return True
+
+
+def player_total(env):
+  sum_gold = 0
+
+  for ent in env.realm.players.values():
+    sum_gold += ent.gold.val
+
+  return sum_gold 
+
+
+def count_actions(tick, actions):
+  cnt_move = 0
+  cnt_attack = 0
+  cnt_sell = 0
+  cnt_use = 0
+  cnt_give = 0
+  cnt_buy = 0
+
+  for entID in list(actions.keys()):
+    for atn, args in actions[entID].items():
```

Comment:
can this just be a dict rather than a long if statment?

Reply:
Got rid of the long if statements.